src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs

```diff
@@ -418,7 +418,7 @@ public Task ManyConcurrentWriteAsyncs()
                         }
                     } while (!tokenSource.Token.IsCancellationRequested);
                 }
-            } while (DateTime.UtcNow - testStartTime <= testRunTime);
+            } while (DateTime.UtcNow - testStartTime <= testRunTime && totalBytesWritten < 10000000);
```

Comment:
where is `totalBytesWritten ` assigned?

Reply:
Actually, I'm not sure this even addresses the concern. All the writing is happening in the inner loop. By the time we get here, we've written everything to the file we're going to write, and when we loop around the outer loop, it overwrites the file with a new empty one to start again.

I can see the value of this condition, but i think for it to be meaningful it needs to be in the inner loop. And then some we're using a length limitation, we can remove the time limitation enabled by the CancellationToken.

Reply:
Nit, 10_000_000 is easier to read than 10000000

Reply:
Well the CancelationToken would still provide value -- it means we need not write all the bytes, and it randomizes somewhat. But fine.